### DISCUSSION TITLE
Random prolonged disconnections ( hours/days ) with FSL2

### DISCUSSION THREAD

**Discussion by Nb-tst** [j-kaltes/Juggluco#152] [General]
Very often ( I'd say with one in two sensors ) a the app lose the connection and it cannot reconnect for hours o even days.
The nfc scan works, so the sensor is still active.

No solution seems to work, not enabling/disabling BT, not cleaning BT cache, not cleaning Juggluco cache, not terminating and reactivating the sensor, not reeboting the phone.
Then, just as ( apparently ) random as it stopped to work, after many hours or even days it reconnects by itself, and start to work as supposed to until the expire date.

Does it happen to someone else? Is it a sensor fault or a smartphone/app issue? 
Has anyone found a way to consistently solve it?

**Comment by j-kaltes** [comment]
On what device do you have these problems?
Did you always have these problems?
Did you have these problems also with Libre 3?
Have you started the Libre 2 sensor with Librelink and is Librelink still active?
Have you used the sensor with other devices?
What does Left menu->Sensor exactly show?
Are there special circumstances in which you have these problems?
Is your phone rooted?

**Reply by Nb-tst** [reply]
- Pixel with android 11
- Not always, but often. I'd say with 40%-50% of sensors.
- I'm not using Libre 3 since a year now. Some of them used to just fail and no nfc scan was possible so I asked for a replacement, though.
- Sensor started with juggluco, no librelink
- No other devices, just this phone
- Status 133
- It may be that when the "faulty" disconnection happens I'm out of range. 99.9% of times I just come back in range and it reconnects instantly, so it may be just a coincidence that usually happens that way.
- No root

**Comment by j-kaltes** [comment]
So the problems are not new.

Pixel 2? https://www.gsmarena.com/google_pixel_2-8733.php

Are battery optimizations disabled for Juggluco?

**Reply by Nb-tst** [reply]
Pixel 2
Battery optimization is disabled both for Juggluco and Bluetooth.

**Reply by Nb-tst** [reply]
Connection restarted yesterday in the late evening, 9 days after it stopped. I can't figure an explanation, maybe it's somehow related with the solar storm that hit Europe in those hours.
